src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5533,91 +5533,102 @@ address MacroAssembler::arrays_equals(Register a1, Register a2, Register tmp3,
       }
     }
   } else {
-    Label NEXT_DWORD, SHORT, TAIL, TAIL2, STUB,
-        CSET_EQ, LAST_CHECK;
+    Label NEXT_64BYTE, NEXT_16BYTE, SHORT, TAIL, TAIL03, TAIL01;
```

Comment:
We'd better not change the default behavior. Make the new code under guard of UseCompactObjectHeaders.

Reply:
> We'd better not change the default behavior. Make the new code under guard of UseCompactObjectHeaders.

done. and also add simd check.